compiler-plugin/src/main/kotlin/arrow/meta/quotes/Quote.kt

```diff
@@ -386,9 +387,10 @@ fun ArrayList<KtFile>.replaceFiles(file: KtFile, newFile: List<KtFile>) {
 
 fun CompilerContext.changeSource(file: KtFile, newSource: String, rootFile: KtFile, sourcePath: String? = null): KtFile {
   var virtualFile = rootFile.virtualFile
-  if (file.name != DEFAULT_META_FILE_NAME) {
-    val path = sourcePath ?: System.getProperty("arrow.meta.generated.source.output", DEFAULT_SOURCE_PATH)
-    val directory = Paths.get(path).toFile()
+  if (sourcePath != null) {
+    val baseDir = configuration?.get(ArrowMetaConfigurationKeys.GENERATED_SRC_OUTPUT_DIR, listOf(DEFAULT_BASE_DIR.toString()))?.get(0)
+    val path = Paths.get(baseDir, sourcePath)
```

Comment:
`baseDir` here is nullable, not sure if it can cause any problems.. maybe we could wrap it with `?.let`

Reply:
Good catch!! Thank you so much @danimontoya :raised_hands: